components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1706,7 +1706,9 @@ std::vector<base::WeakPtr<Tool>> ConversationHandler::GetTools() {
           tools.begin(), tools.end(),
           [&](auto& tool) {
             return (!tool->IsSupportedByModel(model) ||
-                    !tool->SupportsConversation(GetIsTemporary()) ||
+                    !tool->SupportsConversation(
+                        GetIsTemporary(),
+                        associated_content_manager_->HasAssociatedContent()) ||
                     (tool->IsContentAssociationRequired() &&
```

Comment:
We can remove `Tool::IsContentAssociatedRequired` now, but happy to do it in a follow-up if you want

Reply:
seems better to leave it to another PR, let me know if you would like me to create an issue for this